src/Tasks/Common/NETSdkMessage.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Provides a localizable mechanism for logging messages with different levels of importance from the SDK targets.
+    /// </summary>
+    public class NETSdkMessage : NETSdkBaseMessage
+    {
+        public string Importance
+        {
+            get
+            {
+                return MessageImportance;
```

Comment:
How about moving Log.LogWarning, Log.LogError, and Log.LogMessage out to separate `protected abstract void LogMessage(string message)`. Then the base doesn't need to know about importance and you don't need this dance. It also factors the switch into smaller overrides.